rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -181,7 +207,29 @@ void Player::play_messages_until_queue_empty(const PlayOptions & options)
   while (message_queue_.try_dequeue(message) && rclcpp::ok()) {
     std::this_thread::sleep_until(
       start_time_ + std::chrono::duration_cast<std::chrono::nanoseconds>(
-        1.0 / rate * message.time_since_start));
+        1.0 / rate * message.time_since_start + paused_duration_));
```

Comment:
why do we sleep longer here?

Reply:
The `paused_duration_` accounts for the wall duration of the pause. Otherwise, if you have a 10-second bag, and you `ros2 bag play --paused`, wait for 10 seconds, then resume, then the bag will immediately play all the messages at once when you resume.